src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -230,87 +230,66 @@ public virtual void Execute()
                 timeout = (int)Test.Properties.Get(PropertyNames.Timeout);
 
 #if SILVERLIGHT || NETCF
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0)
+            if (Test is TestMethod)
                 RunTestOnOwnThread(timeout);
             else
                 RunTest();
 #elif PORTABLE
             RunTest();
 #else
-            ApartmentState currentApartment = Thread.CurrentThread.GetApartmentState();
+            currentApartment = Thread.CurrentThread.GetApartmentState();
 
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0 || currentApartment != TargetApartment && TargetApartment != ApartmentState.Unknown)
+            if (Test is TestMethod)
```

Comment:
The overall plan for the new framework was documented very completely - or at least verbosely. :-) It seems to be lost now. We have this page: https://github.com/nunit/nunit/wiki/Framework-Design but the pages to which it points are no longer in the wiki. Too bad. If I get time I will try to recover them.

However, I can summarize...

In NUnit V2, the Test object executed itself. There was no overall control of execution.

In 3.0, we created the dispatcher to control test execution. The idea was as follows:

1. Identify special requirements for each test (timeout, STA, etc.)
2. Break each test into work items, each of which can be managed independently
3. Create queues and workers that can handle each special requirement
4. Allow dynamic expansion and contraction of the queues and workers
5. Eliminate the "old" style of running, as exemplified by SimpleWorkItemDispatcher

Like many things we do, this has taken a while with the result that some of it is forgotten, at least by some people. I thnk this is still the way we want to go.

What is done so far (before your fix):
1. We no longer needed a per-test thread for running in the STA, when using the parallel dispatcher.

2. We still needed a per-test thread for timeouts, because we can't yet restart the worker threads dynamically. That's a prerequisite if the worker thread is what gets cancelled by a timeout.

3. We still support RequiresThread, although we had talked about eliminating it.

4. We still support SimpleWorkItemDispatcher, which is used on certain platforms. We could eliminate it but it would require some work and would only have an internal effect.